backend/location/utility.py

```diff
@@ -43,79 +50,80 @@ def get_legacy_location(location_object):
-            exact_address=optional_attributes["exact_address"],
-            country=location_object["country"],
-            name=location_object["name"],
-            centre_point=switched_point,
-            is_formatted=True,
-        )
-        # Postcode location do not have an osm_id
-        if "osm_id" in location_object:
-            loc.osm_id = location_object["osm_id"]
-        loc.save()
-        return loc
+        centre_point = Point(coords[1], coords[0])
+
     elif location_object["type"] == "LineString":
         centre_point = Point(
             float(location_object["lat"]), float(location_object["lon"])
         )
-        loc = Location.objects.create(
-            place_id=location_object["place_id"],
-            city=optional_attributes["city"],
-            state=optional_attributes["state"],
-            place_name=optional_attributes["place_name"],
-            exact_address=optional_attributes["exact_address"],
-            country=location_object["country"],
-            name=location_object["name"],
-            centre_point=centre_point,
-            is_formatted=True,
-        )
-        return loc
-    elif location_object["type"] == "global":
-        loc = get_global_location()
-        return loc
-    else:
+
+    elif (
+        location_object["type"] == "Polygon"
+        or location_object["type"] == "MultiPolygon"
+    ):
         multipolygon = get_multipolygon_from_geojson(location_object["geojson"])
         centre_point = Point(
             float(location_object["lat"]), float(location_object["lon"])
         )
-        loc = Location.objects.create(
-            osm_id=location_object["osm_id"],
-            place_id=location_object["place_id"],
-            city=optional_attributes["city"],
-            state=optional_attributes["state"],
-            place_name=optional_attributes["place_name"],
-            exact_address=optional_attributes["exact_address"],
-            country=location_object["country"],
-            name=location_object["name"],
-            multi_polygon=multipolygon,
-            is_formatted=True,
-            centre_point=centre_point,
-        )
-        return loc
+    else:
+        raise Exception("Unsupported location type")
+
+    loc = Location.objects.create(
+        osm_id=location_object["osm_id"],
+        osm_type=_osm_type_char(location_object["osm_type"]),
+        osm_class=location_object["osm_class"],
+        osm_class_type=location_object["osm_class_type"],
+        place_id=location_object["place_id"],
+        city=location_object["city"],
+        state=location_object["state"],
+        place_name=location_object["place_name"],
+        display_name=location_object["display_name"],
+        exact_address=location_object["exact_address"],
+        country=location_object["country"],
+        name=location_object["name"],
+        centre_point=centre_point,
+        multipolygon=multipolygon,
+        is_formatted=True,
+    )
+
+    loc.save()
```

Comment:
Redundant .save() call: After Location.objects.create() on line 118, calling .save() on line 131 is redundant. The create() method already persists the object to the database.